src/compas_viewer/components/slider.py

```diff
@@ -2,176 +2,79 @@
 from typing import Optional
 
 from PySide6.QtCore import Qt
-from PySide6.QtWidgets import QGridLayout
 from PySide6.QtWidgets import QHBoxLayout
 from PySide6.QtWidgets import QLabel
 from PySide6.QtWidgets import QSlider
+from PySide6.QtWidgets import QVBoxLayout
 from PySide6.QtWidgets import QWidget
 
-from compas.colors import Color
-from compas_viewer.base import Base
-
-
-class Slider(QWidget, Base):
-    """Class representing a horizontal slider wrapped in a grid layout with two rows.
-
-    Parameters
-    ----------
-    action : callable
-        The callback connected to the slide action.
-    value : int, optional
-        The initial value of the slider.
-        Defaults to 0.
-    min_value : int, optional
-        The minimum value of the sliding range.
-        Defaults to 0.
-    max_value : int, optional
-        The maximum value of the sliding range.
-        Defaults to 100.
-    step : int, optional
-        Size of value increments.
-        Defaults to 1.
-    title : str, optional
-        Title label.
-    annotation : str, optional
-        Value annotation label.
-    interval : int, optional
-        The tick interval size.
-        Defaults to 1.
-    bgcolor : :class:`compas.colors.Color`, optional
-        Background color of the box containing the slider.
-    stretch : int, optional
-        Stretch factor of the slider in the grid layout.
-        Defaults to 0.
-    kwargs : dict, optional
-        Additional keyword arguments for the action.
-
-    Attributes
-    ----------
-    action : callable
-        Action associated with the button click event.
-    slider : QtWidgets.QSlider
-        The actual slider widget.
-    value : float
-        The current value of the slider.
-    STYLE : str
-        Stylesheet for the visual appearance of the groove and handle of the slider.
-
-    See Also
-    --------
-    :class:`compas_viewer.layout.SidedockLayout`
-
-    References
-    ----------
-    :PySide6:`PySide6/QtWidgets/QSlider`
-
-    """
-
-    STYLE = """
-    QSlider::groove::horizontal {
-        border: 1px solid #cccccc;
-        background-color: #eeeeee;
-        height: 4px;
-    }
-    QSlider::handle:horizontal {
-        background-color: #ffffff;
-        border: 1px solid #cccccc;
-        border-radius: 6px;
-        height: 12px;
-        width: 12px;
-        margin: -6px 0;
-    }
-    """
 
+class Slider(QWidget):
     def __init__(
         self,
-        action: Callable,
-        value: int = 0,
-        min_value: int = 0,
-        max_value: int = 100,
-        step: int = 1,
-        title: Optional[str] = None,
-        annotation: Optional[str] = None,
-        interval: int = 1,
-        bgcolor: Optional[Color] = None,
-        stretch: int = 0,
-        **kwargs,
+        title: str = "Slider",
+        horizontal: Optional[bool] = True,
+        min_val: float = 0,
+        max_val: float = 100,
```

Comment:
There should be a parameter of `stepsize`, and is it possible to set fraction steps like 0.5? If not then the min and max value type should be int `instead` of `float`

Reply:
I see the boundary we set it to `int` and they define the step